telos-frontend/src/components/todo/DashboardTodo.jsx

```diff
@@ -36,73 +40,168 @@ const useStyles = makeStyles((theme) => ({
     flexWrap: 'wrap',
   },
   datetextField: {
-    marginLeft: theme.spacing(1),
     marginRight: theme.spacing(1),
-    width: 200,
   },
 }));
 
-const listitems = [
-  {
-    name: 'OnGoing',
-    due: '18/03/2021',
-    onGoing: true,
-  },
-  {
-    name: 'OutDated',
-    due: '16/03/2021',
-    onGoing: false,
-  },
-];
-
 const outdated = {
   color: 'red',
 };
 
 const DashboardTodo = () => {
+  const listitems = [
+    {
+      name: 'OnGoing',
+      due: '2021-04-30',
+      onGoing: true,
+      completed: false,
+    },
+    {
+      name: 'OutDated',
+      due: '2021-01-30',
+      onGoing: false,
+      completed: true,
+    },
+    {
+      name: '一二三四五',
+      due: '2021-04-30',
+      onGoing: true,
+      completed: true,
+    },
+    {
+      name: '上山打老虎',
+      due: '2021-04-30',
+      onGoing: true,
+      completed: false,
+    },
+  ];
+
   const classes = useStyles();
-  const [checked, setChecked] = useState([0]);
   const [item, setItem] = useState('');
   const [dueDate, setDueDate] = useState('2021-01-01');
   const [newItem, setNewItem] = useState(listitems);
   const [anchorEl, setAnchorEl] = useState(null);
+  const [cancel, setCancel] = useState([0]);
+  const [reDate, setReDate] = useState('');
+  const [reItem, setReItem] = useState('');
+  const [add, setAdd] = useState(false);
+  const [migrate, setMigrate] = useState(false);
 
-  const handleToggle = (value) => () => {
-    const currentIndex = checked.indexOf(value);
-    const newChecked = [...checked];
+  function dateToTimestamp(endTime) {
+    const date = new Date();
+    date.setFullYear(endTime.substring(0, 4));
+    date.setMonth(endTime.substring(5, 7) - 1);
+    date.setDate(endTime.substring(8, 10));
+    return Date.parse(date) / 1000;
+  }
 
-    if (currentIndex === -1) {
-      newChecked.push(value);
-    } else {
-      newChecked.splice(currentIndex, 1);
+  const sortEvent = () => {
+    const sorted = newItem.sort((a, b) => {
+      return dateToTimestamp(a.due) - dateToTimestamp(b.due);
+    });
+    setNewItem(sorted);
+  };
+
+  const reDateChange = (event) => {
```

Comment:
Functionality looks good, however if you could make some of the naming a bit more clear or add some comments to explain what different things do that would be useful, as I'm not sure what things like this particular function represent